jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/BlockingArrayQueue.java

```diff
@@ -93,8 +91,21 @@ public BlockingArrayQueue()
      */
     public BlockingArrayQueue(int maxCapacity)
     {
-        _elements = new Object[maxCapacity];
-        _growCapacity = -1;
+        this(maxCapacity, maxCapacity, true);
+    }
+
+    /**
+     * Create a bounded {@link BlockingArrayQueue} with given initial and max capacities.
+     *
+     * @param capacity the initial capacity
+     * @param maxCapacity the maximum capacity
+     * @param grow unused
```

Comment:
It should be used: BAQ can be bounded, but also unbounded depending on the value of this parameter.

Reply:
I removed this confusing ctor, as per our discussion.